src/System.Management.Automation/namespaces/RegistryProvider.cs

```diff
@@ -1825,8 +1825,19 @@ public void GetProperty(
                     notePropertyName = LocalizedDefaultToken;
                 }
 
-                propertyResults.Properties.Add(new PSNoteProperty(notePropertyName, key.GetValue(valueName)));
-                valueAdded = true;
+                try
+                {
+                    propertyResults.Properties.Add(new PSNoteProperty(notePropertyName, key.GetValue(valueName)));
+                    valueAdded = true;
+                }
+                catch (InvalidCastException invalidCast)
+                {
+                    WriteError(new ErrorRecord(
+                        invalidCast,
+                        invalidCast.GetType().FullName,
+                        ErrorCategory.WriteError, 
```

Comment:
We read a value. Is there better error category?

Reply:
@iSazonov Maybe `ErrorCategory.InvalidOperation` would be better? I guess I was just following the same pattern as other parts of code when dealing with InvalidCastException.

Reply:
I see  there is `ErrorCategory.ReadError` in the file.

Reply:
Fixed in 44205c999baad1507e931bd7744f81d7970c9c55